src/test_l1_deepSleepMgr.c

```diff
@@ -505,45 +489,32 @@ void test_l1_deepSleepMgr_negative_PLAT_DS_SetDeepSleep (void)
  * |:--:|---------|----------|--------------|-----|
  * |01|Call PLAT_DS_INIT() - open interface | | DEEPSLEEP_SUCCESS | Should Pass |
  * |02|Call PLAT_DS_GetLastWakeupReason() - get last wakeup reason | *wakeupReason | DEEPSLEEPMGR_SUCCESS | Should Pass |
- * |03|Call PLAT_DS_GetLastWakeupReason() - get last wakeup reason | *wakeupReason | DEEPSLEEPMGR_SUCCESS | Should Pass |
- * |04|Verify that the returns from PLAT_DS_GetLastWakeupReason are the same | |  | Should Pass |
- * |05|Call PLAT_DS_TERM() - close interface | | DEEPSLEEP_SUCCESS | Should Pass |
+ * |03|Call PLAT_DS_TERM() - close interface | | DEEPSLEEP_SUCCESS | Should Pass |
  * 
  */
 void test_l1_deepSleepMgr_positive_PLAT_DS_GetLastWakeupReason (void)
 {
+    int result = DEEPSLEEPMGR_SUCCESS;
+    DeepSleep_WakeupReason_t wakeupReason;
     gTestID = 9;
     UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
-	// Define expected results for each variation/step
-    const DeepSleep_Return_Status_t expectedInit = DEEPSLEEPMGR_SUCCESS;
-    const DeepSleep_Return_Status_t expectedGetReason = DEEPSLEEPMGR_SUCCESS;
-    const DeepSleep_Return_Status_t expectedTerm = DEEPSLEEPMGR_SUCCESS;
-
-    // Placeholder for the wakeup reason
-    DeepSleep_WakeupReason_t wakeupReason1;
-    DeepSleep_WakeupReason_t wakeupReason2;
 
     // Variation 01: Call PLAT_DS_INIT() and check the result
-    DeepSleep_Return_Status_t resultInit = PLAT_DS_INIT();
-    UT_ASSERT_EQUAL(resultInit, expectedInit);  
+    result = PLAT_DS_INIT();
+    UT_ASSERT_EQUAL_FATAL(result, DEEPSLEEPMGR_SUCCESS);
 
     // Variation 02: Call PLAT_DS_GetLastWakeupReason() and check the result
-    DeepSleep_Return_Status_t resultGetReason = PLAT_DS_GetLastWakeupReason(&wakeupReason1);
-    UT_ASSERT_EQUAL(resultGetReason, expectedGetReason);
-
-    // Variation 03: Call PLAT_DS_GetLastWakeupReason() and check the result
-    DeepSleep_Return_Status_t resultGetReason2 = PLAT_DS_GetLastWakeupReason(&wakeupReason2);
-    UT_ASSERT_EQUAL(resultGetReason2, expectedGetReason);
-
-    // Variation 04: Verify that the returns are the same
-    UT_ASSERT_EQUAL(wakeupReason1, wakeupReason2);
+    result = PLAT_DS_GetLastWakeupReason(&wakeupReason);
+    PLAT_DS_ASSERT_AUTO_TERM_NUMERICAL(result, DEEPSLEEPMGR_SUCCESS);
+    UT_LOG("\n %s  wakeupReason=%d \n",__FUNCTION__,wakeupReason);
 
     // At this point, you may want to further verify if 'wakeupReason' contains valid data.
     // This is dependent on your specific requirements and what PLAT_DS_GetLastWakeupReason() actually does.
```

Comment:
this comment can be removed

Reply:
keeping this comment for feature reference